src/HvLightPipe.c

```diff
@@ -20,11 +20,13 @@
 #include <xmmintrin.h>
 #define hv_sfence() _mm_sfence()
 #elif __arm__ || HV_SIMD_NEON
-  #if __ARM_ACLE
+  #ifdef __ARM_ACLE
     #include <arm_acle.h>
     // https://msdn.microsoft.com/en-us/library/hh875058.aspx#BarrierRestrictions
     // http://doxygen.reactos.org/d8/d47/armintr_8h_a02be7ec76ca51842bc90d9b466b54752.html
     #define hv_sfence() __dmb(0xE) /* _ARM_BARRIER_ST */
+  #elif defined ARM_CORTEX
+    #define hv_sfence() __ASM volatile ("dmb 0xF":::"memory")
```

Comment:
__ARM_ACLE is not defined by gcc and __dmb is not defined in gcc's arm_acle.h.
I took `dmb 0xF` from the definition of __DMB in CMSIS (core_cmInstr.h line 484):
`/** \brief  Data Memory Barrier
    This function ensures the apparent order of the explicit memory operations b
efore and after the instruction, without ensuring their completion.
 */
__attribute__((always_inline)) __STATIC_INLINE void __DMB(void)
{
  __ASM volatile ("dmb 0xF":::"memory");
}
`
Maybe 0xF would be better as 0xE if that is how you use it in this context (write only barrier).